operator/controllers/keptnevaluation/controller.go

```diff
@@ -0,0 +1,344 @@
+	"github.com/keptn-sandbox/lifecycle-controller/operator/api/v1alpha1/semconv"
+)
+
+// KeptnEvaluationReconciler reconciles a KeptnEvaluation object
+type KeptnEvaluationReconciler struct {
+	client.Client
+	Scheme   *runtime.Scheme
+	Recorder record.EventRecorder
+	Log      logr.Logger
+	Meters   common.KeptnMeters
+	Tracer   trace.Tracer
+}
+
+//+kubebuilder:rbac:groups=lifecycle.keptn.sh,resources=keptnevaluations,verbs=get;list;watch;create;update;patch;delete
+//+kubebuilder:rbac:groups=lifecycle.keptn.sh,resources=keptnevaluations/status,verbs=get;update;patch
+//+kubebuilder:rbac:groups=lifecycle.keptn.sh,resources=keptnevaluations/finalizers,verbs=update
+//+kubebuilder:rbac:groups=lifecycle.keptn.sh,resources=keptnevaluationproviders,verbs=get;list;watch
+//+kubebuilder:rbac:groups=lifecycle.keptn.sh,resources=keptnevaluationdefinitions,verbs=get;list;watch
+
+// Reconcile is part of the main kubernetes reconciliation loop which aims to
+// move the current state of the cluster closer to the desired state.
+// TODO(user): Modify the Reconcile function to compare the state specified by
+// the KeptnEvaluation object against the actual cluster state, and then
+// perform operations to make the cluster state reflect the state specified by
+// the user.
+//
+// For more details, check Reconcile and its Result here:
+// - https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.12.2/pkg/reconcile
+func (r *KeptnEvaluationReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	r.Log.Info("Reconciling KeptnEvaluation")
+	evaluation := &klcv1alpha1.KeptnEvaluation{}
+
+	if err := r.Client.Get(ctx, req.NamespacedName, evaluation); err != nil {
+		if errors.IsNotFound(err) {
+			// taking down all associated K8s resources is handled by K8s
+			r.Log.Info("KeptnEvaluation resource not found. Ignoring since object must be deleted")
+			return ctrl.Result{}, nil
+		}
+		r.Log.Error(err, "Failed to get the KeptnEvaluation")
+		return ctrl.Result{Requeue: true, RequeueAfter: 30 * time.Second}, nil
+	}
+
+	traceContextCarrier := propagation.MapCarrier(evaluation.Annotations)
+	ctx = otel.GetTextMapPropagator().Extract(ctx, traceContextCarrier)
+
+	ctx, span := r.Tracer.Start(ctx, "reconcile_evaluation", trace.WithSpanKind(trace.SpanKindConsumer))
+	defer span.End()
+
+	semconv.AddAttributeFromEvaluation(span, *evaluation)
+
+	if !evaluation.IsStartTimeSet() {
+		// metrics: increment active evaluation counter
+		r.Meters.AnalysisActive.Add(ctx, 1, evaluation.GetActiveMetricsAttributes()...)
+		evaluation.SetStartTime()
+	}
+
+	if evaluation.Status.RetryCount >= evaluation.Spec.Retries {
+		r.recordEvent("Warning", evaluation, "ReconcileTimeOut", "retryCount exceeded")
+		err := fmt.Errorf("RetryCount for evaluation exceeded")
+		span.SetStatus(codes.Error, err.Error())
+		evaluation.Status.OverallStatus = common.StateFailed
+		r.updateFinishedEvaluationMetrics(ctx, evaluation, span)
+
+		return ctrl.Result{}, err
```

Comment:
not sure if we should return an error if the evaluation was not successful -> setting its status to Failed is fine of course, but I would not see it as an error in the reconciler